addons/website/views/website_templates.xml

```diff
@@ -145,6 +145,17 @@
         <link rel="preconnect" href="https://fonts.gstatic.com/" crossorigin=""/>
     </xpath>
 
+    <xpath expr="//head/script" position="before">
+        <t t-if="website and website.cookies_bar and website.block_third_party_domains and not request.env.user.has_group('website.group_website_designer') and not request.env['ir.http']._is_allowed_cookie('optional') and not editable">
+            <script src="/website/static/src/js/content/cookie_watcher.js"/>
+            <script>
+                watch3rdPartyScripts(
+                    `<t t-out='website.default_blocked_third_party_domains + "\n" + website.sudo().blocked_third_party_domains'/>`.split("\n")
+                );
```

Comment:
Mmh, I though using the session info did not require to use any Odoo tools but they changed that :man_facepalming: 

Anyways:
> // This script is to be called directly in the <head>. It should not import any
// other file or library, as it cannot be transformed into a module (because a
// module is deferred): the function should be the very first thing launched on
// the page to be able to bypass other IIFE of 3rd-party services that inject
// new scripts client-side.

Your script is transformed into a module since https://github.com/odoo/odoo/commit/5f2c505836002ac7851c29c28d612f721d467bc4
So either something is fundamentally wrong because of it, or you can just make it part of the non-lazy loaded bundle. I don't see why it could not be part of the assets_frontend_minimal bundle: it should not be delayed doing that, JS loading is done synchronously. To confirm.

Reply:
Bundles are deferred (I'm not talking about lazy-loading):
`<script type="text/javascript" defer="defer" src="/web/assets/1/04d7222/web.assets_frontend_minimal.min.js" onerror="__odooAssetError=1"></script>`

If I'm not wrong, https://github.com/odoo/odoo/commit/5f2c505836002ac7851c29c28d612f721d467bc4 made sure that, _in bundles_, files are considered as odoo modules by default.
Here, as I call the file directly, it's not considered a module (neither an Odoo one, nor an ES6 one), which is why it works.